topic1/src/keyPoint6/CreditCardMethod.java

```diff
@@ -14,12 +15,12 @@ public CreditCardMethod(String nm, String ccNumer){
 	}
 
 	@Override
-	public void pay(ArrayList<ShoppingCartLine> shoppingCartLineCollection) {
+	public void pay(List<ShoppingCartLine> shoppingCartLineCollection) {
 		System.out.println("paid with credit card --> 10% discount");
 		int sum = 0;
 		for (int index = 0; index < shoppingCartLineCollection.size() ; index++){
```

Comment:
You can use 
`for(ShoppingCartLine` line : shoppingCartLineCollection) {`
to avoid dealing with indexes.